backend/prisma/schema.prisma

```diff
@@ -32,8 +54,111 @@ model Session {
 }
 
 model Profile {
-  id     String @id @default(uuid())
-  user   User @relation(fields: [userId], references: [id])
-  userId String @unique @default(uuid())
-  photoUrl String? 
-}
\ No newline at end of file
+  id        String   @id @default(uuid())
+  user      User     @relation(fields: [userID], references: [id])
+  userID    String   @unique
+  photoUrl  String?
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+}
+
+model Plan {
+  id               String       @id @default(uuid())
+  description      String
+  startDate        DateTime
+  endDate          DateTime
+  createdDate      DateTime     @default(now())
+  specialEventType EventType
+  contact          Contact?     @relation(fields: [contactID], references: [id])
+  contactID        String?      @unique
+  User             Pledge[]
+  createdAt        DateTime     @default(now())
+  updatedAt        DateTime     @updatedAt
+  GiftChoice       GiftChoice[]
+}
+
+model Pledge {
+  id           String        @id @default(uuid())
+  plan         Plan          @relation(fields: [planID], references: [id])
+  planID       String
+  user         User          @relation(fields: [userID], references: [id])
+  userID       String
+  inviteStatus InviteStatus?
+  role         Role
+  amount       Int?
+  pledgeStatus PledgeStatus?
+  createdAt    DateTime      @default(now())
+  updatedAt    DateTime      @updatedAt
+}
+
+enum PledgeStatus {
+  SENT_MONEY
+  HAVENT_SENT_MONEY
+}
+
+enum Role {
+  PLAN_LEADER
+  FRIEND
+}
+
+enum InviteStatus {
+  INVITED
+  RESENT_INVITE
+  ACCEPTED
+  DENY
+  NOT_APPLICABLE
+}
+
+model Contact {
+  id                 String                    @id @default(uuid())
+  email              String?
+  firstName          String
+  notes              String?
+  source             String
+  owner              User                      @relation(fields: [ownerID], references: [id])
+  ownerID            String
+  importantDateEvent ImportantDateEvent[]
+  relationship       UserContactRelationship[]
+  Plan               Plan?
```

Comment:
I'm not super well versed in Prisma so i'd defer to @justinpchen94 here. 

Don't we want to save the plan ID here and shouldn't this be a 1 Contact to Many Plan relationship too? Each contact will eventually have more than one plan associated with them (i.e. their second birthday on the system).

How are we modeling that right now?

Reply:
What I had in mind was that only 1 contact can 0 or 1 plan. We can do 1 Contact to Many Plan.

Reply:
![image](https://user-images.githubusercontent.com/106122976/225109878-3888c3d9-3dc5-4631-b931-87beeed3381f.png)